Add downloadURL config option

arudge opened the pull request:
Accepts a valid java.net.URI for mongodb download

arudge commented:
Addressed the comments from https://github.com/sourcemuse/GradleMongoPlugin/pull/27

roberttaylor426 commented on build.gradle:
```diff
@@ -6,12 +6,14 @@ apply plugin: 'signing'
 targetCompatibility = 1.7
 
 repositories {
+    mavenLocal()
```
Not keen to add a dependency to mavenLocal(), we want builds to be repeatable. Would you mind dropping this?

roberttaylor426 commented on build.gradle:
```diff
@@ -6,12 +6,14 @@ apply plugin: 'signing'
 archivesBaseName = 'gradle-mongo-plugin'
 group = 'com.sourcemuse.gradle.plugin'
-version = '0.9.0-SNAPSHOT'
+version = '0.9.1-SNAPSHOT'
```
0.9.0 hasn't been released yet, so would you mind leaving the version as-is?

(I'm happy to cut a release as soon as this PR is merged).

roberttaylor426 commented on src/test/groovy/com/sourcemuse/gradle/plugin/GradleMongoPluginExtensionSpec.groovy:
```diff
@@ -123,4 +123,25 @@ class GradleMongoPluginExtensionSpec extends Specification {
+
+    def 'mongod download url throws exception for invalid url'() {
+        given:
+        String invalidURL = 'thisisnotavalidurl'
```
Tiny comment, but would you mind keeping the spacing between the `given` and `when` blocks? It just keeps things consistent with the other tests and helps readability.

roberttaylor426 commented:
A couple of small comments, otherwise LGTM.

roberttaylor426 commented on build.gradle:
```diff
@@ -6,12 +6,14 @@ apply plugin: 'signing'
 
 repositories {
+    mavenLocal()
+    jcenter()
```
Why do we need this?

arudge commented:
Thanks for the review!
I addressed all your concerns, but let me know if I missed one.

Adding `jcenter()` allows dependency resolution from 'https://bintray.com/'. 
Nothing critical, so I removed it. 

If you want to create some Github issues for any new features or house keeping items, I could contribute some additional pull requests.

roberttaylor426 commented on src/test/groovy/com/sourcemuse/gradle/plugin/GradleMongoPluginExtensionSpec.groovy:
```diff
@@ -123,4 +123,27 @@ class GradleMongoPluginExtensionSpec extends Specification {
+
+    def 'mongod download url can be set for valid url'() {
+        given:
+        String invalidURL = 'http://google.com'
```
`validUrl` surely? :-)

arudge replied:
bah, I really shouldn't try to get out PRs while kids are running around.

roberttaylor426 commented:
Thanks Andy that would be great. I was planning to address #28 by the end of this week, but if you want to beat me to it please feel free. :-)

One more small comment added btw.

arudge commented:
Ok, should be good now
